src/main/java/org/apache/camel/idea/completion/extension/YamlPropertyPlaceholdersSmartCompletion.java

```
@@ -0,0 +1,130 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.idea.completion.extension;
+
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.PlainPrefixMatcher;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.camel.idea.completion.extension.CamelPropertiesSmartCompletionExtension.IGNORE_PROPERTIES;
+
+/**
+ * Smart completion for editing a Camel endpoint uri, to show a list of YAML properties can be added.
+ * For example editing <tt>jms:queue?{{_CURSOR_HERE_</tt>. Which presents the user
+ * with a list of possible properties.
+ */
+public class YamlPropertyPlaceholdersSmartCompletion implements CamelPropertyCompletion {
+
+    @NotNull
+    private Map<String, Object> getProperties(VirtualFile virtualFile) {
+        Map<String, Object> result = new HashMap<>();
+        File file = new File(virtualFile.getPath());
+        Yaml yaml = new Yaml();
+
+        try {
+            InputStream ios = new FileInputStream(file);
+            // Parse the YAML file and return the output as a series of Maps and Lists
+            result = (Map<String, Object>) yaml.load(ios);
+        } catch (Exception e) {
+            e.printStackTrace();
```

We should probably just ignore this error. Or find out how to use IDEA logger and then log to idea log instead.